tests/has/devfile_source.go

```diff
@@ -131,6 +144,49 @@ var _ = framework.HASSuiteDescribe("devfile source", func() {
 		Expect(route.Spec.Host).To(Not(BeEmpty()))
 		klog.Infof("Component route host: %s", route.Spec.Host)
 	})
+
+	It("Check GitOpsDeployment component deployment health", func() {
+		Eventually(func() bool {
+			deployment, _ := framework.CommonController.GetAppDeploymentByName(QuarkusComponentName, AppStudioE2EApplicationsNamespace)
```

Comment:
What happen if GetAppDeploymentByName returns an error?
Check the error:
if the deployment don't exist wait and if it is different error return false and the tests will fail